### ISSUE TITLE
⚙️ Update the basic input and textarea components

### ISSUE THREAD

**Issue by substrae** [GCTC-NTGC/gc-digital-talent#6739]
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/84e3db07-fce3-4e37-9945-c7c70308974b)

## 🌟 Details ([Figma file](https://www.figma.com/file/T1R64oNWog10OXiBnuiMkz/2.-Components?type=design&node-id=758-5888))

These two input elements have been updated, though the changes to "required" fields can be propagated to all inputs if you have the time.

### Accessibility

- Placeholder text should only exist for standard inputs and not textareas; it should be concise and provide a concrete example of the expected data, not instruction
- `aria-describedby` should be used to indicate whether context or error messages exist
- Because unsaved changes are only indicated by a subtle border change, this state should be announced to AT
- Textareas should also indicate their word limit; @esizer I defer to you on how this is implemented
  - `aria-describedby` can accept multiple, space separated, id's so that should be the best way to indicate this [es]

### Typography

- Input labels should use the `caption` font-size
- Input placeholders and values should use the `copy` font-size
  - Placeholder text should use the lightest gray possible to meet contrast
  - Value text should use `black`
- Both context messages and error messages should use the `caption` font-size
- A textarea's word counter should use the `caption` font-size

### Size

Note that all inputs should use the `rounded` border-radius value.

### Standard inputs

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/9521d5d3-062a-404d-a416-09b9c9b2bc80)

#### Aligning buttons

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/cf2f4e51-eaa5-4d34-aab9-303adf50db22)

### Textareas

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/2a4a9410-900d-4bac-a291-9455caba936a)

### Required fields

- Required fields have been simplified to meet web standards; no more "Optional" and "Required" text
- Optional fields have no marker
- Required fields are marked with an `*` character

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/b3b5c129-3a85-42b8-9ac0-c89abc6c6931)

### Extra context

- Context wells have been updated to show at all times (compared to the current toggle) for both better accessibility and clearer access to the information
- These wells should use the colors from default wells, but be sure to use the `caption` font-size value for text

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/27e6b6ab-4a2d-4774-9d56-e425208cb263)

### Textarea word counts

- Word counts are optional, but when they are added, they appear inside the textarea element but on top of padding that prevents the input's content from ever overlapping the counter
- The counter should use `black` by default, but change to `error.darker` when the count is exceeded

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/196b9df6-efc1-44d7-924b-9e2265802750)

## 🔄️ States

- By default, an input's border should use `gray`

### Unsaved changes

- Border changes to `secondary.darker`

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/cda724c9-1e37-43d0-a790-c7701613ffc7)

### Validation error

- Border changes to `error.darker`
- Concise, human-readable error well appears (border: `error.darker`, background: `error.lightest`)

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/6960386/6bfdae49-13c5-4add-a2cf-fe749c1ca452)
